cdap-watchdog/src/main/java/co/cask/cdap/logging/write/LogCleanup.java

```diff
@@ -109,13 +121,127 @@ public Void call() throws Exception {
     }
   }
 
+  private void cleanFilesWithoutMeta(final long tillTime, final Map<String, NamespaceId> namespacedLogBaseDirMap,
+                                     final SetMultimap<String, Location> parentDirs) throws Exception {
+    LOG.info("Starting deletion of log files older than {} without metadata", tillTime);
+    List<NamespaceMeta> namespaceMetaList = namespaceQueryAdmin.list();
```

Comment:
That means we will never cleanup logs in impersonated namespace. Which I think should be accepted since cdap does not privilege on the directory it cannot do anything in it.